mne/io/write.py

```diff
@@ -250,6 +262,7 @@ def end_file(fid):
     fid.write(np.array(FIFF.FIFFT_VOID, dtype='>i4').tostring())
     fid.write(np.array(data_size, dtype='>i4').tostring())
     fid.write(np.array(FIFF.FIFFV_NEXT_NONE, dtype='>i4').tostring())
+    check_fiff_length(fid)
```

Comment:
This is the only check that *needs* to be there, but I put in a few others in places that big data chunks could be written to try to save people some pain.